### PR TITLE
feat(react-headless-components-preview): add CompoundButton

### PR DESCRIPTION
**Pull request by mainframev**
## Previous Behavior

The headless preview package had Button, MenuButton, and ToggleButton primitives, but no CompoundButton subpath.

## New Behavior

- Adds `@fluentui/react-headless-components-preview/compound-button` with stable component, hook, renderer, props, slots, and state exports.
- Exposes `data-disabled`, `data-disabled-focusable`, `data-icon-only`, and `data-has-secondary-content` as presence/absence state selectors (`''` when present, omitted when false).
- Adds conformance, semantics, interaction, keyboard, accessibility, slot, override, and edge-case tests.
- Adds the export map, API report, bundle fixture, token-only CSS Module stories, docs, and a zero-semver `patch` change file.

## Validation

- Headless tests, lint, type-check, build, generate-api, and bundle-size
- React 17/18 integration coverage
- Story lint, type-check, Storybook build, SSR, and public docsite build
- Playwright verification for interaction, disabled inverse, focus, light/dark tokens, RTL, and docs

## Stack

1. #36529 — regression coverage
2. #36530 — expose CompoundButton base APIs
3. This PR

### CODE REVIEW

File: packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
+import * as React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import { isConformant } from '../../testing/isConformant';
+import { CompoundButton } from './CompoundButton';
+
+expect.extend(toHaveNoViolations);
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx]: q: could we add this to headless `isConformant` helper to ensure all components are covered?

File: packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    const { getByRole } = render(
+      <CompoundButton as="a" href="#compound-button">
+        Link
+      </CompoundButton>,
+    );
+
+    expect(getByRole('link', { name: 'Link' })).toHaveAttribute('href', '#compound-button');
+  });
+
+  it('forwards its ref to the root', () => {
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx]: should be already covered by conformance tests

File: packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+  it('invokes click handlers when enabled', () => {
+    const onClick = jest.fn();
+    const { getByRole } = render(<CompoundButton onClick={onClick}>Enabled</CompoundButton>);
+
+    fireEvent.click(getByRole('button'));
+
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('activates a focused enabled button with Enter', () => {
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx]: it's a default browser behavior, not something custom we added, so we might not  need to test it

File: packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    const button = getByRole('button');
+
+    button.focus();
+    userEvent.keyboard('{Enter}');
+
+    expect(button).toHaveFocus();
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('activates a focused enabled button with Space', () => {
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx]: same as above

File: packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    const button = getByRole('button');
+
+    button.focus();
+    userEvent.keyboard('{space}');
+
+    expect(button).toHaveFocus();
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('blocks activation and emits an exact disabled presence attribute when disabled', () => {
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.test.tsx:82]: same here

File: packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.types.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import type {
+  CompoundButtonBaseProps,
+  CompoundButtonBaseState,
+  CompoundButtonSlots as CompoundButtonBaseSlots,
+} from '@fluentui/react-button';
+
+/**
+ * CompoundButton component slots.
+ */
+export type CompoundButtonSlots = CompoundButtonBaseSlots;
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/library/src/components/CompoundButton/CompoundButton.types.ts]: pls convert to inline re-exports

File: packages/react-components/react-headless-components-preview/stories/src/CompoundButton/compound-button.module.css [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+.iconAction[data-icon-only] {
+  display: inline-flex;
+  min-inline-size: var(--space-16);
+  inline-size: var(--space-16);
+  padding: 0;
+  justify-content: center;
+}
+
+@media (prefers-reduced-motion: reduce) {
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/stories/src/CompoundButton/compound-button.module.css]: Do we actually need this for SB docs?

File: packages/react-components/react-headless-components-preview/stories/src/CompoundButton/CompoundButtonDescription.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+A compound button presents a primary action with optional secondary text that adds concise context. Both text lines contribute to the accessible name, so keep them brief and avoid repeating the same information.
+
+`CompoundButton` renders a native button by default and can render an anchor with `as="a"` and `href`. A disabled button uses the native `disabled` attribute and cannot be focused or activated. A disabled-focusable button remains in the tab order, exposes `aria-disabled="true"`, and blocks activation.
+
+The root exposes presence-based state attributes. Each attribute is present with the exact value `""` when its state is true and is omitted when false:
+
+- `data-disabled`
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/stories/src/CompoundButton/CompoundButtonDescription.md]: q: do we want to document `data-attributes` in the component description or to have a dedicated section below the props table similar to the https://github.com/microsoft/fluentui/pull/36024/changes

Comment (mainframev) [reply packages/react-components/react-headless-components-preview/stories/src/CompoundButton/CompoundButtonDescription.md]: added, but perhaps the best way would be to have that in automated way similar to prop tables

File: packages/react-components/react-headless-components-preview/stories/src/CompoundButton/compound-button.module.css [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+.iconAction[data-icon-only] {
+  display: inline-flex;
+  min-inline-size: var(--space-16);
+  inline-size: var(--space-16);
+  padding: 0;
+  justify-content: center;
+}
+
+@media (forced-colors: active) {
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/stories/src/CompoundButton/compound-button.module.css]: not sure we needs this for headless docsite

File: packages/react-components/react-headless-components-preview/stories/src/CompoundButton/CompoundButtonDescription.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
+A compound button presents a primary action with optional secondary text that adds concise context. Both text lines contribute to the accessible name, so keep them brief and avoid repeating the same information.
+
+`CompoundButton` renders a native button by default and can render an anchor with `as="a"` and `href`. A disabled native button uses the `disabled` attribute and cannot be focused or activated. A disabled anchor cannot use a native disabled attribute, so it exposes `aria-disabled="true"`, removes `href`, blocks activation, and still emits the root `data-disabled` attribute. Disabled native buttons also emit root `data-disabled`. A disabled-focusable button remains in the tab order, exposes `aria-disabled="true"`, and blocks activation.
+
+The `contentContainer` slot groups the primary and secondary text, while `secondaryContent` represents the second textual line. These semantic slots let consumers apply their own layout and typography without changing the component's behavior.
+
+## State attributes
```
Comment (dmytrokirpa) [inline packages/react-components/react-headless-components-preview/stories/src/CompoundButton/CompoundButtonDescription.md:7]: probably fine for now, but we need a generic solution for all components to extract and render those below the component props table

**Review by dmytrokirpa — APPROVED** [review]
